spacy/pipeline/textcat.py

```diff
@@ -72,7 +72,7 @@
     "textcat",
     assigns=["doc.cats"],
     default_config={
-        "threshold": 0.5,
+        "threshold": 0.0,
```

Comment:
I think `0.0` as default is fine - either way can be somewhat misleading, depending on whether this is `multi_label`or not.

Reply:
`textcat` should never be `multi_label` (well, unless you've hacked a bunch of things into meaninglessness). `multi_label=False` is hard-coded in the default scorer.

Reply:
Ah, right. Then I'm in favor of `None` as default `threshold` value for `textcat`, but I'm ok with `0` as well.

Reply:
I changed it to `Optional[float]` for `textcat` but it makes the joint API docs tricky, so I'm still not sure if we want to do this in the end.

Reply:
Let's stick with `0` if it's too much of a bother. The docstring documents this sufficiently well, I think.